app/Module/GoogleMapsModule.php

```diff
@@ -1077,158 +917,144 @@ function createMarker(point, name, html, mhtml, icontype) {
 			},
 			streetViewControl:        false,  // Show Pegman or not
 			scrollwheel:              true
-		};
-		var pm_map = new google.maps.Map(document.querySelector(".gm-map"), myOptions);
-		google.maps.event.addListener(pm_map, "click", function() {
-			jQuery(".gm-ancestor.person_box")
-				.removeClass("person_box")
-				.addClass("gm-ancestor-visited");
-			infowindow.close();
-			lastlinkid = null;
 		});
-		// create the map bounds
-		var bounds = new google.maps.LatLngBounds();';
-		// add the points
-		$curgen       = 1;
-		$count        = 0;
-		$colored_line = array(
-			'1' => '#FF0000',
-			'2' => '#0000FF',
-			'3' => '#00FF00',
-			'4' => '#FFFF00',
-			'5' => '#00FFFF',
-			'6' => '#FF00FF',
-			'7' => '#C0C0FF',
-			'8' => '#808000',
-		);
-		$lat        = array();
-		$lon        = array();
-		$latlongval = array();
-		for ($i = 0; $i < $this->treesize; $i++) {
-			// moved up to grab the sex of the individuals
-			$person = $this->ancestors[$i];
-			if ($person) {
-				$name = $person->getFullName();
-
-				// -- check to see if we have moved to the next generation
-				if ($i + 1 >= pow(2, $curgen)) {
-					$curgen++;
-				}
 
-				$relationship = FunctionsCharts::getSosaName($i + 1);
+		var oms = new OverlappingMarkerSpiderfier(map, {
+			markersWontMove: true,
+			markersWontHide: true,
+			nearbyDistance:  5
+		});
+		var point;
+		var marker;
+		jQuery.each(<?php echo json_encode($marker_data); ?>, function(index, item) {
+			point = new google.maps.LatLng(item.lat, item.lon);
+			bounds.extend(point);
+			//Create a marker
+			var baseIcon = new google.maps.MarkerImage(item.iconimg);
+			marker = new google.maps.Marker({
+				icon:       baseIcon,
+				map:        map,
+				position:   point,
+				title:      item.name,
+				html:       item.html,
+				zIndex:     100 + index,
+				id:         item.id,
+				baseicon:   baseIcon,
+				name:       item.name
+			});
+			oms.addMarker(marker);
+			gmarkers[item.id] = marker;
+		});
 
-				// get thumbnail image
-				if ($person->getTree()->getPreference('SHOW_HIGHLIGHT_IMAGES')) {
-					$image = $person->displayImage();
-				} else {
-					$image = '';
-				}
+		var line;
+		var plines;
+		var colors     = ['#000000', '#FF0000', '#0000FF', '#00FF00', '#FFFF00', '#00FFFF', '#FF00FF', '#C0C0FF', '#808000'];
+		jQuery.each(<?php echo json_encode($link_lines); ?>, function(index, item) {
+			line   = [
+				new google.maps.LatLng(item.parent_lat, item.parent_lon),
+				new google.maps.LatLng(item.child_lat, item.child_lon)
+			];
+			plines = new google.maps.Polygon({
+				paths:         line,
+				strokeColor:   colors[item.generation],
```

Comment:
If `generation` is more than the size of the colors array, this will error.
Perhaps use something like `colors[item.generation % colors.length]`